src/components/Header.tsx

```diff
@@ -1,39 +1,128 @@
-import React from 'react';
-import { useSelector } from 'react-redux'
+import React, {useState} from 'react';
+import { useSelector, connect } from 'react-redux'
 import {Block, Text} from '../components';
+
 import { TouchableOpacity } from 'react-native-gesture-handler';
 import IoniconsIcon from 'react-native-vector-icons/Ionicons';
-import {Image, SafeAreaView, StyleSheet, Button, View} from 'react-native';
+import FeatherIcon from 'react-native-vector-icons/FontAwesome';
+import {Image, SafeAreaView, StyleSheet, Button, View, ViewStyle, StyleProp} from 'react-native';
 import styles from '../assets/styles/styles'
-import {types, actions} from '../store';
-
+import {types, actions, reducers, store} from '../store';
+import {Picker} from 'react-native';
+import { useDispatch } from 'react-redux'
+import Menu, { MenuItem, MenuDivider } from 'react-native-material-menu';
+import { RootState } from '../store/types';
+import style from '../assets/styles/styles';
+const useThunkDispatch = () => useDispatch<typeof store.dispatch>();
 export interface headerProps {
     pageName : string,
     backNavigationCB ?: () => any;
     user ?: types.user.User;
+    style ?: StyleProp<ViewStyle>,
+    children ?: React.ReactNode,
+    [index: string]: any;
 }
 
 const Header = (props : headerProps) => {
-    const { pageName, backNavigationCB, user } = props;
-    return  <Block row style={{paddingVertical: 15}}>
-    {
-    backNavigationCB ? (
-        <TouchableOpacity activeOpacity={0.8} onPress={backNavigationCB} style={{...styles.back}}> 
-            <IoniconsIcon name="ios-arrow-round-back" size={45} color="white"/>            
-        </TouchableOpacity>
-    ) : null
-    }
-      <Block flex={2.5} style={{flexShrink : 2}} center>
-      <Text numberOfLines={2} h3 white style={{fontSize:20, marginRight: 3}}>
+  const dispatch = useThunkDispatch();
+
+  let _menu : Menu;
+  const setMenuRef = (ref : Menu) => {
+    _menu = ref;
+  };
+ 
+  const hideMenu = () => {
+    _menu.hide();
+  };
+ 
+  const showMenu = () => {
+    _menu.show();
+  };
+
+  const signOut = async () => {
+    dispatch<any>(actions.user.logoutUser()) // huh this is dumb
+    hideMenu();
+  }
+  const { pageName, backNavigationCB, user, style, children, ...rest} = props;
+  return  (
+    <Block flex={0.42} column style={{style, paddingHorizontal: 15, }} {...rest} >
+      <Block flex={false} row style={{paddingVertical: 15}}>
+      {
+      backNavigationCB ? (
+          <TouchableOpacity activeOpacity={0.8} onPress={backNavigationCB} style={{...styles.back}}> 
+              <IoniconsIcon name="ios-arrow-round-back" size={45} color="white"/>            
+          </TouchableOpacity>
+      ) : null
+      }
+      <Block center flex={1} >
+        <Text h3 white style={{fontSize:21}} numberOfLines={1}>
```

Comment:
The page name is pushed to the left by the user profile picture by 35, so make sure to add  marginLeft:35 to the text property so the page name is centered.